main.go

```diff
@@ -0,0 +1,261 @@
+		} `json:"type"`
+	} `json:"types"`
+}
+
+func main() {
+	config := &Config{Cache: pokecache.NewCache(10 * time.Second), Pokedex: make(map[string]Pokemon)}
+	s := bufio.NewScanner(os.Stdin)
+	for {
+		fmt.Print("Pokedex > ")
+		s.Scan()
+		input := s.Text()
+		args := cleanInput(input)
+		if len(args) == 0 {
+			continue
+		}
+		switch args[0] {
+		case "exit":
+			commandExit()
+			return
+		case "help":
+			fmt.Println("Welcome to the Pokedex!\nUsage:\n\nhelp: Displays a help message\nexit: Exit the Pokedex")
+		case "map":
+			mapCommand(config)
+		case "mapb":
+			mapbCommand(config)
+		case "explore":
+			if len(args) < 2 {
+				fmt.Println("Usage: explore <area_name>")
+				continue
+			}
+			exploreCommand(config, args[1])
+		case "catch":
+			if len(args) < 2 {
+				fmt.Println("Usage: catch <pokemon_name>")
+				continue
+			}
+			catchCommand(config, args[1])
+		case "inspect":
+			if len(args) < 2 {
+				fmt.Println("Usage: inspect <pokemon_name>")
+				continue
+			}
+			inspectCommand(config, args[1])
+		default:
+			fmt.Println("Unknown command")
+		}
+	}
+}
+
+func mapCommand(config *Config) {
+	url := config.Next
+	if url == "" {
+		url = "https://pokeapi.co/api/v2/location-area/"
+	}
+
+data, err := fetchLocations(url, config.Cache)
+	if err != nil {
+		fmt.Println("Error fetching locations:", err)
+		return
+	}
+	for _, location := range data.Results {
+		fmt.Println(location.Name)
+	}
+	config.Next = data.Next
+	config.Previous = data.Previous
+}
+
+func mapbCommand(config *Config) {
+	url := config.Previous
+	if url == "" {
+		fmt.Println("you're on the first page")
+		return
+	}
+    data, err := fetchLocations(url, config.Cache)
+	if err != nil {
+		fmt.Println("Error fetching locations:", err)
+		return
+	}
+	for _, location := range data.Results {
+		fmt.Println(location.Name)
+	}
+	config.Next = data.Next
+	config.Previous = data.Previous
+}
+
+func exploreCommand(config *Config, location string) {
+	url := fmt.Sprintf("https://pokeapi.co/api/v2/location-area/%s/", location)
+	fmt.Printf("Exploring %s...\n", location)
+	data, err := fetchExplore(url, config.Cache)
+	if err != nil {
+		fmt.Println("Error fetching location details:", err)
+		return
+	}
+	fmt.Println("Found Pokemon:")
+	for _, encounter := range data.PokemonEncounters {
+		fmt.Printf(" - %s\n", encounter.Pokemon.Name)
+	}
+}
+
+func catchCommand(config *Config, name string) {
+	url := fmt.Sprintf("https://pokeapi.co/api/v2/pokemon/%s/", name)
+	fmt.Printf("Throwing a Pokeball at %s...\n", name)
+	data, err := fetchPokemon(url, config.Cache)
+	if err != nil {
+		fmt.Println("Error fetching Pokemon details:", err)
+		return
+	}
+	rand.Seed(time.Now().UnixNano())
+	catchChance := rand.Intn(data.BaseExp + 1)
+	if catchChance%2 == 0 {
+		fmt.Printf("%s was caught!\n", name)
+		config.Pokedex[name] = *data
+	} else {
+		fmt.Printf("%s escaped!\n", name)
+	}
+}
+
+func fetchLocations(url string, cache *pokecache.Cache) (*LocationResponse, error) {
```

Comment:
все фетч методы кстати можно на дженериках сделать

```suggestion
func fetch[T any](url string, cache *pokecache.Cache) (*T, error) {
	if cachedData, found := cache.Get(url); found {
		var data T
		if err := json.Unmarshal(cachedData, &data); err == nil {
			return &data, nil
		}
	}
	
	resp, err := http.Get(url)
	if err != nil {
		return nil, fmt.Errorf("get data: %w", err)
	}
	defer resp.Body.Close()
	
	var data T
	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
		return nil, fmt.Errorf("decode response: %w", err)
	}
	
	responseData, err := json.Marshal(data)
	if err != nil {
		return &data, nil
	}
	cache.Add(url, responseData)
	return &data, nil
}

func fetchLocations(url string, cache *pokecache.Cache) (*LocationResponse, error) {
	return fetch[LocationResponse](url, cache)
}

func fetchPokemon(url string, cache *pokecache.Cache) (*Pokemon, error) {
	return fetch[Pokemon](url, cache)
}

func fetchExplore(url string, cache *pokecache.Cache) (*ExploreResponse, error) {
	return fetch[ExploreResponse](url, cache)
}
```